haystack/nodes/document_classifier/transformers.py

```diff
@@ -151,14 +155,13 @@ def predict(self, documents: List[Document], batch_size: Optional[int] = None) -
             for doc in documents
         ]
         batches = self.get_batches(texts, batch_size=batch_size)
-        if self.task == "zero-shot-classification":
-            batched_predictions = [
-                self.model(batch, candidate_labels=self.labels, truncation=True) for batch in batches
-            ]
-        elif self.task == "text-classification":
-            batched_predictions = [
-                self.model(batch, return_all_scores=self.return_all_scores, truncation=True) for batch in batches
-            ]
+        batched_predictions = []
+        for batch in tqdm(batches, disable=not self.progress_bar, desc="Classifying"):
```

Comment:
I think `self.get_batches` will return a generator, so `tqdm` won't really show a progress bar nut the number of executed iterations. I think by setting the `total` argument, we get a progress bar.